desktop-widgets/mainwindow.cpp

```diff
@@ -1835,9 +1822,15 @@ void MainWindow::showFilterIfEnabled()
 		ui.bottomSplitter->setSizes({ EXPANDED, COLLAPSED });
 	}
 }
-void MainWindow::registerApplicationState(ApplicationState state, QWidget *topLeft, QWidget *topRight, QWidget *bottomLeft, QWidget *bottomRight)
+void MainWindow::registerApplicationState(ApplicationState state, QWidget *topLeft, QWidget *topRight, QWidget *bottomLeft, QWidget *bottomRight,
+					  int topLeftFlags, int topRightFlags, int bottomLeftFlags, int bottomRightFlags)
```

Comment:
I know this is a weird question... but what makes more sense:
`(state, widget1, widget2, widget3, widget4, flag1, flag2, flag3, flag4)`
or
`(state, widget1, flag1, widget2, flag2, widget3, flag3, widget4, flag4)`
?
For some reason I first thought the second one would be better. But after writing this, I'm not sure anymore. So consider this food for thought...

Reply:
First I was also going with `state, widget` pairs. But then I thought these lines:
```
        registerApplicationState(ApplicationState::Default, mainTab.get(), profileContainer, diveList, mapWidget,
                                                            FLAG_NONE, FLAG_NONE, FLAG_NONE, FLAG_NONE);
```
were more in line with the old code. I'm perfectly fine with changing it.

Reply:
Well, I just realized that when going the (widget, flag) way, we can arrange the quadrants in the source code like they are on the screen. Pure genius - let's do this. 😀

Reply:
💯